### ISSUE TITLE
Bug migration script doesn't warn anout PHP 8

### ISSUE THREAD

**Issue by daraelmin** [Dolibarr/dolibarr#24790]
### Bug

When upgrading to 18.0 (and 17.0 ?), from 15.0 to 16.0, the installation script is ok with PHP 8.1 but throw an error as PHP 8 was not supported for these version.

Fatal error: Uncaught mysqli_sql_exception: Unknown column 'u.flagdelsessionsbefore' in 'field list' in htdocs/core/db/mysqli.class.php:371 Stack trace: #0 htdocs/core/db/mysqli.class.php(371): mysqli_result->fetch_object() #1 tdocs/install/upgrade2.php(4561): DoliDBMysqli->fetch_object(Object(mysqli_result)) #20398 htdocs/install/upgrade2.php(485): migrate_user_photospath() #3 {main} thrown in htdocs/core/db/mysqli.class.php on line 371

### Environment Version

16.0

### Environment PHP

PHP 8

### Expected and actual behavior

Expected behavior, the migration script should warn that the version of PHP should be PHP 7 until Dolibarr v17

**Comment by ksar-ksar** [comment]
Hello,

Sorry but I don't understand.

If you want to update to V17 from whatever version, you will put V17 PHP files that are compatible with PHP V8.
So what do you try to achieve ?

Also the error message, doesn't seems to be linked to PHP version but a mismatch between PHP and MYSQL ?

**Comment by daraelmin** [comment]
Hello @ksar-ksar ,

The typical situation is when you use an old version, and you decide to upgrade to the last one:

For example, say you're on dolibarr v9 with PHP 5.4 (v9 was 5.3+) and you decide to go to 17.0.2.

When copying the file of the v17 on your system and run the installation script, it will warn you to use a more recent version of PHP, at least the 5.6. So you decide to update your PHP version to PHP 8. Now the script says all is okay. But dolbarr v10 didn't support PHP 8, neighter for PHP 7. So your migration crash.

Cheers.

**Comment by ksar-ksar** [comment]
Hello,

This not suppose to be possible.
When you upload V17, you have the V17 migration script that do all the upgrades V10 to V17.
V17 migration script is compatible with PHP V8.

Do you have the error message?

**Comment by daraelmin** [comment]
Hello,

The production site was migrated through staging. It took me a while, that's why, I couldn't answer right away, but there you go.

To reproduce the error, I took one of the backups, unfortunately, I did not keep the first ones (v10 to 14), but I reproduced one of these errors.

`Fatal error: Uncaught mysqli_sql_exception: Unknown column 'u.flagdelsessionsbefore' in 'field list' in htdocs/core/db/mysqli.class.php:371 Stack trace: #0 htdocs/core/db/mysqli.class.php(371 ): mysqli_result->fetch_object() #1 htdocs/install/upgrade2.php(4561): DoliDBMysqli->fetch_object(Object(mysqli_result)) #2 htdocs/install/upgrade2.php(485): migrate_user_photospath() #3 { main} thrown in htdocs/core/db/mysqli.class.php on line 371`


This one occurs from v15 to v16 when trying to migrate to v18 with PHP 8.1.

Cheers
